redis.conf

```
@@ -1309,7 +1309,28 @@ disable-thp yes
 
 appendonly no
 
-# The name of the append only file (default: "appendonly.aof")
+# The base name of the append only file (default: "appendonly.aof")
+# 
+# Now we support the MULTI PART AOF mechanism, that is, we will have multiple 
+# AOF files at the same time. They are divided into two types, one is the the 
+# BASE type, it represents the full amount of data (SNAPSHOT) after each AOF 
+# rewrite. There is only one BASE AOF at most. The second is INCR type, they 
+# may have more than one. They represent the incremental commands since the 
+# last AOF rewrite.
+# 
+# In order to manage these AOF files, we use a AOF META file to record them.
+# 
+# Therefore, these files mentioned above must have a certain naming rule. For 
+# compatibility reasons, the original `appendfilename` configuration will be 
+# used as the BASE part of the file name. At the same time, BASE/INCR/META files 
+# have corresponding name suffixes (They are _b_, _i_, _manifest). Since there may 
+# be multiple AOF files, in order to ensure that they do not conflict, we add a 
+# sequence to the end of the file name.
+# 
+# Therefore, the final naming rule is: 
+# BASE AOF: appendonly.aof_b_1、appendonly.aof_b_2 etc.
+# INCR AOF: appendonly.aof_i_1、appendonly.aof_i_2 etc.
+# META FILE: appendonly.aof_manifest
 
 appendfilename "appendonly.aof"
```

being a base file name, i wanna see just name here, and i wanna add the suffix (file type) at runtime.
i don't mind trimming the `.aof` suffix we get from the user (in case he didn't update his config file), and i also don't mind keeping it and adding another suffix after it.
but for new users / deployments, i want to have something that's cleaner, so i suggest:
```suggestion
appendfilename "appendonly"
```

---

If the user configures the name as `appendonly.xxxxx` (not the default `appendonly.aof`), should we also trim it to appendonly?

---

i don't think so. as i said, i'm also willing to keep this suffix, but since it's a common config i figured we can make a minimal effort to clean it. i.e. if the string has at least 5 chars, and ends with that 4 chars suffix, trim the last 4 chars.

but now it occurs to me that maybe if some piece of external software will try to take the value of that config and search for the files based on this base name, this trimming can cause some confusion.
on the other hand, it is likely that this "server management software" is coupled with the deployment scripts and the server configuration, so i don't think it's a real issue.

let's seek additional feedback. @yossigo @soloestoy ?

---

Do we have any conclusions about this? 

Can we directly use the original name (after all, it is configured by the user or default),  and then we use `base``or `incr` to indicate the type of this file, such as `appendonly.aof.2.base`, `appendonly.aof.3.incr`。

The current solution  does not seem to be very good: `appendonly.aof.2.rdb`, `appendonly.aof.3.aof`, it is difficult to understand that a name contains both `rdb` and `aof`.

---

for now, let's keep the simple approach you have (just use it as base, and possibly generate odd file names).
we can focus on other aspects of this PR and leave that one for last.
then i guess we need to map a few cases and what would be the outcome of each to decide.
i.e. users who upgrade from an old system who use the default config, vs users who upgrade from an old system who have explicitly overwritten it, and how will redis behave on upgrade (should load the old data in either one of the cases)

---

OK

---

i see you changed the suffix (saw your last commit), i meant to just use what ever is in the config as base file name and apply the suffixes on it (i.e. that there's no need for now to strip the ".aof" suffix that the user may have provided).
i.e that i'm ok with "appendonly.aof.0.rdb"
maybe i don't mind the base/incr suffix, but i think it is redundant (it's also indicated by the meta file).
what the suffix came to denote in my "design" is the format inside the file (i.e. considering we'll deprecate the possibility for AOFRW to generate AOF command content, the base file is an RDB format).

---

Well, I understand what you mean. One reason for changing `.rdb` to `.base` is because in the current pr, the BASE file may be in AOF or RDB format (depending on whether the `server.aof_use_rdb_preamble` yes or no), if BASE is in AOF format but we still end with `.rdb` may be confusing (the user will try this features of this pr when merged, but the next pr hasn't merged yet ). Note that what I said is only in this pr. If we remove `server.aof_use_rdb_preamble` and `rewriteAppendOnlyFileRio` in the next pr, we can determine that our BASE is rdb and INCR is aof. So  `appendonly.aof.1.rdb` and  `appendonly.aof.2.aof`  will be certain (reflecting their internal format).

This may be my concern. After all our next PR (remove `server.aof_use_rdb_preamble` and `rewriteAppendOnlyFileRio`) should be merged soon. So if we don’t mind this short-term confusing, I am willing to roll back the suffix to the previous version.

@oranagra WDYT？

---

Theoretically, if we want this PR to be really clean on it's own (and not count on the next change), maybe we can either add a `.aof` or `.rdb` suffix for the rewrite depending on `server.aof_use_rdb_preamble` (to indicate the format).
and if additionally we also want some notation if it's a base or incr, we can keep these too.
then maybe in the next pr we can remove this complication.

on the other hand, i don't mind letting it be one way or the other in the short term, and i'm quite sure we're gonna trim the AOF generation code very soon, i'm just worried that we won't forget it and keep only the "base" notation and no indication on the format. so maybe it's better to just add both (not a big overhead in coding)

---

Currently `base` and `incr` use separate seq, they can be incremented separately. If `base` also uses the aof suffix(`server.aof_use_rdb_preamble` is no), there may be a file name conflict between `base` and `incr`. If `base` and `incr` share a seq, then seq is not continuously increasing. @oranagra  Which do you think is better?

---

i guess we can invest a few extra lines of code (which we may clean in the future), and add both "base" / "incr" notation in the file name, as well as ".aof" or ".rdb" to denote the format.
shouldn't be too complicated, and we may consider changing that in a few weeks.

---

@oranagra Do you like this?

`
appendonly.aof_rdb.1.base
appendonly.aof_aof.1.base

appendonly.aof_aof.1.incr
appendonly.aof_aof.2.incr`

---

for some reason i prefer the suffix to denote the format, maybe i'm looking too much into it, but what complications does it add? is it more than some 4 lines of code?

---

I have modified the PR according to the rules mentioned above.

```
appendonly.aof_1.base.rdb
appendonly.aof_1.base.aof

appendonly.aof_1.incr.aof
appendonly.aof_2.incr.aof
```

---

@oranagra PLZ review this part again and see if there are any more problems? thanks.

---

LGTM